src/Agent.js

```diff
@@ -146,6 +149,11 @@ export default class Agent {
 		if (context) {
 			this._applyContext(req, context);
 		}
+		if (Object.keys(headers).length) {
+			for (const key of Object.keys(headers)) {
+				req.set(key, headers[key]);
+			}
+		}
 		if (query) {
 			req.query(query);
 		}
```

Comment:
comparing to https://github.com/particle-iot/particle-api-js/pull/115 which closely duplicates this functionality, i see a couple of key differences:

1. i refactored all the base http methods (`.get()`, `.put()`, etc) to accept an object instead of individual params as i find more than 3 unwieldy
2. i tried to clean up `.createWebhook()` as it has always accepted a headers object and thus conflicted with the new `headers` arg / prop ([see here](https://github.com/particle-iot/particle-api-js/pull/115/commits/4dc6551df1230b61875ae0d59bb9b833a54cd539))
3. i made `.downloadFile()` use `uri` instead of `url` for uniformity

one downside of my approach is that it definitely represents a breaking change so a bump from `8` to `9` is warranted.

i'm on the fence... i'm not sure how disruptive it'd be to go with my refactor as the appeal of the lib isn't its base http methods... but if i had to pick i guess i'd go with my approach if only because of item 2 in my list above. not a super-strongly held position though.

Reply:
actually, there's another even bigger difference between this and #115 - the later extends every high-level method (e.g. `.sendOtp()`, `.createCustomer()`, etc) to optionally accept `headers` which is think is useful (perhaps necessary?)

so, eesh... yeah, another point in favor of adopting #115 i guess. sorry for the churn, let's chat about how best to divide an conquer 👍